span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanPreProcessor.java

```diff
@@ -26,48 +23,27 @@
     implements Transformer<byte[], Span, KeyValue<byte[], PreProcessedSpan>> {
 
   static final String SPANS_COUNTER = "hypertrace.reported.spans";
-  private static final String DROPPED_SPANS_COUNTER = "hypertrace.reported.spans.dropped";
-  private static final String IS_LATE_ARRIVAL_SPANS_TAGS = "is_late_arrival_spans";
-  private static final String LATE_ARRIVAL_THRESHOLD_CONFIG_KEY =
-      "processor.late.arrival.threshold.duration";
   private static final Logger LOG = LoggerFactory.getLogger(JaegerSpanPreProcessor.class);
   private static final ConcurrentMap<String, Counter> statusToSpansCounter =
       new ConcurrentHashMap<>();
-  private static final ConcurrentMap<String, Counter> tenantToSpansDroppedCount =
-      new ConcurrentHashMap<>();
-  private static final ConcurrentMap<String, Counter> tenantToLateArrivalSpansDroppedCount =
-      new ConcurrentHashMap<>();
-  private static final Duration minArrivalThreshold = Duration.of(30, ChronoUnit.SECONDS);
   private TenantIdHandler tenantIdHandler;
-  private SpanFilter spanFilter;
-  private Duration lateArrivalThresholdDuration;
+  private SpanDropManager spanDropManager;
 
   public JaegerSpanPreProcessor() {
     // empty constructor
   }
 
   // constructor for testing
-  JaegerSpanPreProcessor(Config jobConfig) {
+  JaegerSpanPreProcessor(Config jobConfig, ExcludeSpanRuleCache excludeSpanRuleCache) {
     tenantIdHandler = new TenantIdHandler(jobConfig);
-    spanFilter = new SpanFilter(jobConfig);
-    lateArrivalThresholdDuration = configureLateArrivalThreshold(jobConfig);
+    spanDropManager = new SpanDropManager(jobConfig, excludeSpanRuleCache);
```

Comment:
Can we rather instantiate `ExcludeSpanRuleCache` inside `SpanDropManager`, or more preferably? inside `ExcludeSpanRuleEvaluator`